File: 1501.py
file = open('_inputs_/1501.0')
line = file.read().strip()
res = 0
for c in line:
    if c == '(':
        res += 1
    else:
        res -= 1
count = 0
for i in range(len(line)):
    if count < 0:
        res2 = i
        break
    c = line[i]
    if c == '(':
        count += 1 
    else:
        count -= 1
print(res)
print(res2)

File: 1501.js
let fs = require('fs')
let input = fs.readFileSync('_inputs_/1501.1').toString().trim().split('')
let res = input.reduce((floor, c) => c == '(' ? ++floor : --floor, 0)
let floor = 0

// p2 golf
// let res2 = input.map(c => c == '(' ? ++floor : --floor).indexOf(-1) + 1

// p2 naive
floor = 0
i = -1
while (++i < input.length) {
	if (floor < 0)
		break
	floor = input[i] == '(' ? floor + 1 : floor - 1
}
res2 = i
console.log(res, res2)
